src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Frameworks/SupportedTargetFrameworksProvider.cs

```diff
@@ -0,0 +1,105 @@
+﻿// Licensed to the .NET Foundation under one or more agreements. The .NET Foundation licenses this file to you under the MIT license. See the LICENSE.md file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.ComponentModel.Composition;
+using System.Linq;
+using System.Threading.Tasks;
+using System.Threading.Tasks.Dataflow;
+using Microsoft.Build.Framework.XamlTypes;
+using Microsoft.VisualStudio.ProjectSystem.Properties;
+using EnumCollection = System.Collections.Generic.ICollection<Microsoft.VisualStudio.ProjectSystem.Properties.IEnumValue>;
+using EnumCollectionProjectValue = Microsoft.VisualStudio.ProjectSystem.IProjectVersionedValue<System.Collections.Generic.ICollection<Microsoft.VisualStudio.ProjectSystem.Properties.IEnumValue>>;
+
+namespace Microsoft.VisualStudio.ProjectSystem.VS.Frameworks
+{
+    /// <summary>
+    ///     Responsible for producing valid values for the TargetFramework property from a design-time build.
+    /// </summary>
+    [ExportDynamicEnumValuesProvider("SupportedTargetFrameworksEnumProvider")]
+    [AppliesTo(ProjectCapability.DotNet)]
+    internal class SupportedTargetFrameworksProvider : ChainedProjectValueDataSourceBase<EnumCollection>, IDynamicEnumValuesProvider, IDynamicEnumValuesGenerator
+    {
+        private readonly IProjectSubscriptionService _subscriptionService;
+
+        [ImportingConstructor]
+        public SupportedTargetFrameworksProvider(
+            ConfiguredProject project,
+            IProjectSubscriptionService subscriptionService)
+            : base(project, synchronousDisposal: true, registerDataSource: false)
+        {
+            _subscriptionService = subscriptionService;
+        }
+
+        [ConfiguredProjectAutoLoad]
+        [AppliesTo(ProjectCapability.DotNet)]
+        public void Load()
+        {
+            // To avoid UI delays when opening the AppDesigner for the first time, 
+            // we auto-load so that we are included in the first design-time build
+            // for the project.
+            EnsureInitialized();
+        }
+
+        protected override IDisposable? LinkExternalInput(ITargetBlock<EnumCollectionProjectValue> targetBlock)
+        {
+            IProjectValueDataSource<IProjectSubscriptionUpdate> source = _subscriptionService.ProjectBuildRuleSource;
+
+            // Transform the changes from design-time build -> Supported target frameworks
+            DisposableValue<ISourceBlock<EnumCollectionProjectValue>> transformBlock = source.SourceBlock.TransformWithNoDelta(
+                update => update.Derive(Transform),
+                suppressVersionOnlyUpdates: true,
```

Comment:
setting suppressVersionOnlyUpdates to true is almost always wrong other than linking to an end action block. It is usually will lead into dead locks.

Reply:
This is a self-contained block (no one else waits on it other than itself) - are saying if this gets sync'linked with something else?

Reply:
Because this class implements IProjectValueDataSource<EnumCollection>, and it is hard to say that no other code chain to it in the future by reading the code alone.  Yeah, if it is not chained to SyncLink,  no code call GetLatestAsync/GetSpecificVersionAsync to the endpoint, or need to report to the operation progress feature, suppressVersionOnlyUpdates is ok to use.

Reply:
@davkean and @lifengl Can we get a resolution on this today on the CPS meeting? Right now 16.8 is broken and this is last comment we need resolved to fix the experience. We need this ASAP.